### PR TITLE
Add support for building snap packages

### PR DESCRIPTION
**Pull request by chrisglass**
This branch makes a few changes to support running the exo binary from a
snap package (https://snapcraft.io).

The biggest difference is that snaps force the usage of $HOME/.exoscale/ as the source of configuration (a specific hole in the normal snap confinement mechanism was granted for this specific location).

Note: this is not meant to be "production ready" yet, a few rough
corners need to be polished first, but this at least allows manual
publication into the "edge channel".

Testing:

This is already published to the edge channel, so the following on any ubuntu system should just work:

```
sudo snap install --edge exoscale-cli
exo config
# Follow prompt
exo vm list
```

Note: a request for alias was made, so that in the future the installed binary can be "exo".

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,6 +9,7 @@ GO_FILES = $(shell find . -type f -name '*.go')
 .PHONY: all
 all: clean exo
 
+.PHONY: exo
```
Comment (pyr) [inline Makefile]: It shouldn't be phony because it creates an executable called `exo`, should it?

Comment (chrisglass) [reply Makefile]: Good point. Fixed.

**Comment by greut** [comment]
Could the goreleaser config help? https://goreleaser.com/customization/#Snapcraft

**Comment by falzm** [comment]
Good to know for the GoReleaser Snap support, thanks @greut.

**Comment by chrisglass** [comment]
Yes, that is likely the best path forward since goreleaser can push to the snap store. Thanks!

In the mean time we need some paperwork with he snapstore to make sure the user experience is what we expect (read/write in $HOME/.exoscale, having an "exo" automatic alias). 

I guess this PR should have been a draft PR until then :)

**Comment by falzm** [comment]
I'm a bit confused here: isn't one of the `.goreleaser.yml` or `snapcraft.yaml` files redundant, or are both needed? cc @pierre-emmanuelJ

**Comment by chrisglass** [comment]
I guess the goreleaser is supposed to override the snapcraft file - but so far it's not been shown to actually work unfortunately.

Perhaps we could consider to keep the know-working snapcraft file to already publish snaps (by hand or otherwise, in a way that actually works), and consider migrating to goreleaser in another branch?

**Pull request merged**